File: day01/day01.go
package main

import (
        "bufio"
        "fmt"
        "io"
        "os"
        "strconv"
)

func fuelRequired(mass int) int {
        return ((mass / 3) - 2)
}

func main() {
        file, err := os.Open("input")
        if err != nil {
                panic(err)
        }
        
        defer file.Close()
        
        reader := bufio.NewReader(file)
        
        result := 0
        for {
                line, _, err := reader.ReadLine()
                
                if err == io.EOF {
                        break
                }
                
                value, _ := strconv.Atoi(string(line))
                result += fuelRequired(value)
        }
        fmt.Printf("%d \n", result)
}

File: day01/day01.c
#include <stdio.h>

int requiredFuel(int mass)
{
    return mass / 3 - 2;
}

int main()
{
    int mass;
    int total = 0;
    while (scanf("%d", &mass) != EOF)
    {
        total += requiredFuel(mass);
    }
    printf("%d\n", total);
    return 0;
}
